src/analysis/convex/stone_separation.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import analysis.convex.join
+
+/-!
+# Stone's separation theorem
+
+This file prove Stone's separation theorem. This tells us that any two disjoint convex sets can be
+separated by a convex set whose complement is also convex.
+
+It can be seen as a "cheap" version of the geometric Hahn-Banach theorem, in the sense that the
+separating set in Stone's separation corresponds to the halfspace in geometric Hahn-Banach, but in a
+much broader class of spaces (any vector space, instead of real normed spaces).
```

Comment:
What you have is fine but how about:
```suggestion
In the special case that the ambient space carries the structure of a locally convex topological
vector space (over ℝ), the Hahn-Banach separation theorems provide stronger statements: one may find
a separating hyperplane (subject to appropriate hypotheses).
```
(assuming I've got that right)

Reply:
I quite liked "cheap" but your version is more descriptive.